### ISSUE TITLE
QP code path: solver-supported Constraint classes aren't being checked

### ISSUE THREAD

**Issue by rileyjmurray** [cvxpy/cvxpy#2154]
The QpSolver base class (from which OSQP and others inherit) supports Zero cone and NonPos cone constraints. If we define a problem explicitly with a NonNeg constraint and then request OSQP as a solver, then I would expect the QP canonicalization code path to either (1) convert it to NonPos or (2) raise an error saying that OSQP doesn't support NonNeg constraints.

Neither of these things is happening right now; here is a minimal breaking example.
```
import cvxpy as cp
import numpy as np
from cvxpy.tests.base_test import BaseTest

class MyTest(BaseTest):

    def test_nonneg(self):
        n = 3
        x = cp.Variable(n)
        c = np.arange(n)
        prob = cp.Problem(cp.Minimize(cp.sum(x)), [cp.NonNeg(x - c)])

        # Solve through cone program path.
        prob.solve(solver=cp.ECOS)
        self.assertItemsAlmostEqual(x.value, c)

        # Solve through QP path.
        prob.solve(solver=cp.OSQP)  # OSQP says the problem is unbounded!
        self.assertItemsAlmostEqual(x.value, c)
```
Small note: I discovered this while modifying CVXPY to reduce dependency on the NonPos class in favor of NonNeg. There is a ``test_nonpos`` function in ``test_constraints.py`` that's almost identical to the function I wrote above. I expected that test to fail because OSQP isn't declared as supporting NonNeg, but it actually failed because OSQP claimed it was handed an unbounded problem.

**Comment by rileyjmurray** [comment]
Could someone familiar with QP canonicalization look into this? Ping @akshayka, @SteveDiamond, and @bstellato.

**Comment by rileyjmurray** [comment]
I think the solution is to do something like this commit:
https://github.com/cvxpy/cvxpy/pull/2155/commits/6fb2f915d6828503f50e9d57106a9b717476b0a6
More specifically, we'd need to create and use a new ``nonneg2nonpos`` function, rather than use the existing ``nonpos2nonneg`` function as I did in that commit.

**Comment by SteveDiamond** [comment]
Good catch! I agree with your proposed fix.

**Comment by rileyjmurray** [comment]
resolved by #2155
